flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/JsonFunctionsTest.scala

```diff
@@ -117,12 +131,48 @@ class JsonFunctionsTest extends ExpressionTestBase {
       s"$candidate is not json scalar")
 
     for (sql <- sqlCandidates) {
-      try {
-        testSqlApi(sql, "null")
-      } catch {
+      try testSqlApi(sql, "null") catch {
         case e: Exception => assertEquals(e.getClass, expectedException)
       }
     }
   }
 
+  @Test
+  def testJsonExists(): Unit = {
+    testSqlApi("json_exists(f9, '$.info.type')", "false")
+
+    // lax json test
+    testSqlApi("json_exists(f9, 'lax $')", "true")
+    testSqlApi("json_exists(f9, 'lax $.info.type')", "true")
+    testSqlApi("json_exists(f9, 'lax $.info.address.town')", "true")
+    testSqlApi("json_exists(f9, 'lax $.info.\"address\"')", "false")
+    testSqlApi("json_exists(f9, 'lax $.info.tags')", "true")
+    testSqlApi("json_exists(f9, 'lax $.info.type[0]')", "false")
+    testSqlApi("json_exists(f9, 'lax $.info.none')", "false")
+    testSqlApi("json_exists(f9, 'lax $' error on error)", "true")
+
+    // strict + no error
+    testSqlApi("json_exists(f9, 'strict $.info.type')", "true")
+    testSqlApi("json_exists(f9, 'strict $.info.type1')", "false")
+    // strict + error
+    testSqlApi("json_exists(f9, 'strict $.info.type1' false on error)", "false")
+    testSqlApi("json_exists(f9, 'strict $.info.type1' true on error)", "true")
+    testSqlApi("json_exists(f9, 'strict $.info.type' true on error)", "true")
+    testSqlApi("json_exists(f9, 'strict $.info.type' false on error)", "true")
+    testSqlApi("json_exists(f9, 'strict $.info.type' unknown on error)", "true")
+    testSqlApi("json_exists(f9, 'strict $.info.\"type\"' unknown on error)", "null")
+    testSqlApi("json_exists(f9, 'strict $.info.type' error on error)", "true")
+
+    verifyJsonExistsException("json_exists(f7, 'lax aa')", "SQL validation failed")
+  }
+
+  private def verifyJsonExistsException[T <: Exception](
+       sqlExpr: String,
+       expectedMessage: String
+     ): Unit = {
+    expectedException.expect(classOf[ValidationException])
+    expectedException.expectMessage(startsWith(expectedMessage))
+    testSqlApi(sqlExpr, "null")
+    expectedException = ExpectedException.none()
```

Comment:
have you verified this test? I seems that if we encounter one exception, it will abort the test no matter it satisfies the expected exception.

Reply:
> have you verified this test? I seems that if we encounter one exception, it will abort the test no matter it satisfies the expected exception.

Well, I tested this place. no problem.

Reply:
> have you verified this test? I seems that if we encounter one exception, it will abort the test no matter it satisfies the expected exception.

you are right, Well,  I have changed it.